### DISCUSSION TITLE
Help with manual property changed

### DISCUSSION THREAD

**Discussion by schragnasher** [AvaloniaUI/Avalonia#13176] [Q&A]
I am using ReactiveUI and trying to get a property changed event to fire properly. I think this should work and I am fairly sure if i manually implemented NotifyPropertyChange it would, but I want to use RxUI.

I have a class

```
public class Foo {

   public int Bar { get; set; } = 1;

}
```
And a ViewModel

```
public class MyViewModel : ViewModelBase
{

    [Reactive]
    public Foo MyFoo { get; set; }

    private void OnSelectionChanged()
    {
        // I have a view that binds to MyFoo.Bar
        MyFoo.Bar= 2;
        this.RaisePropertyChanged(nameof(MyFoo ));
    }
}
```
When OnSelectionChanged is called it changes the value of Bar. As bar is not reactive, and I cant make it reactive I want to force my View to update. Calling RaisePropertyChanged on MyFoo does not cause this to happen. Unless i actually Change the value of MyFoo itself first to a new Foo instance, and also remove Reactive from MyFoo. Is there a way to make this work?

**Comment by timunie** [comment]
you need to make Foo to implement INotifyPropertyChanged (e.g. make it ReactiveObject) and notify inside for Bar.

**Reply by schragnasher** [reply]
"you need to make Foo to implement INotifyPropertyChanged"

Lets say that is impossible. Maybe its from an external library.

**Reply by thevortexcloud** [reply]
Then you can't do it. You can fake it though. EG Wrap the object with another class that implements all the properties and what not and also the interface. Which will just raise the event and passthrough/return any values to/from the backing object,

Although that wont help you if the backing object updates on its own.
